static/js/settings_profile_fields.js

```diff
@@ -66,15 +66,27 @@ function delete_profile_field(e) {
     update_profile_fields_table_element();
 }
 
-function read_select_field_data_from_form(field_elem) {
+function read_select_field_data_from_form(field_elem, old_field_data) {
     const field_data = {};
     let field_order = 1;
+
+    const old_option_value_map = new Map();
+    if (old_field_data !== undefined) {
+        for (const [value, choice] of Object.entries(old_field_data)) {
+            old_option_value_map.set(choice.text, value);
+        }
+    }
+
     $(field_elem)
         .find("div.choice-row")
         .each(function () {
             const text = $(this).find("input")[0].value;
             if (text) {
-                field_data[field_order - 1] = {text, order: field_order.toString()};
+                if (old_option_value_map.get(text) !== undefined) {
+                    $(this).attr("data-value", old_option_value_map.get(text));
```

Comment:
This code is added to make sure that in a case when I delete one existing option, but add same option again before saving and then save the changes, then user who had already selected that value retain their selection.

Reply:
OK, let's move this into a comment, since it's pretty subtle.